x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+      });
+
+      it('should set `activeApiTokenIsExisting` to false when the provided token has no id', () => {
+        mount({
+          activeApiTokenIsExisting: true,
+        });
+        const { id, ...newTokenWithoutId } = newToken;
+        CredentialsLogic.actions.toggleCredentialsForm(newTokenWithoutId);
+        expect(CredentialsLogic.values.activeApiTokenIsExisting).toEqual(false);
+      });
+
+      it('should set `activeApiTokenIsExisting` to false when no token is provided', () => {
+        mount({
+          activeApiTokenIsExisting: true,
+        });
+        CredentialsLogic.actions.toggleCredentialsForm();
+        expect(CredentialsLogic.values.activeApiTokenIsExisting).toEqual(false);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.toggleCredentialsForm(newToken);
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiTokenIsExisting: expect.any(Boolean),
+        activeApiToken: expect.any(Object),
+        activeApiTokenRawName: expect.any(String),
+        formErrors: expect.any(Array),
+        showCredentialsForm: expect.any(Boolean),
+      });
+    });
+  });
+
+  describe('hideCredentialsForm', () => {
+    describe('activeApiTokenRawName', () => {
+      it('resets this value', () => {
+        mount({
+          activeApiTokenRawName: 'foo',
+        });
+        CredentialsLogic.actions.hideCredentialsForm();
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual('');
+      });
+    });
+    describe('showCredentialsForm', () => {
+      it('resets this value', () => {
+        mount({
+          showCredentialsForm: true,
+        });
+        CredentialsLogic.actions.hideCredentialsForm();
+        expect(CredentialsLogic.values.showCredentialsForm).toEqual(false);
+      });
+    });
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.hideCredentialsForm();
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        showCredentialsForm: expect.any(Boolean),
+        activeApiTokenRawName: expect.any(String),
+      });
+    });
+  });
+
+  describe('resetCredentials', () => {
+    describe('isCredentialsDetailsComplete', () => {
+      it('should reset to false', () => {
+        mount({
+          isCredentialsDetailsComplete: true,
+        });
+        CredentialsLogic.actions.resetCredentials();
+        expect(CredentialsLogic.values.isCredentialsDetailsComplete).toEqual(false);
+      });
+    });
+    describe('isCredentialsDataComplete', () => {
+      it('should reset to false', () => {
+        mount({
+          isCredentialsDataComplete: true,
+        });
+        CredentialsLogic.actions.resetCredentials();
+        expect(CredentialsLogic.values.isCredentialsDataComplete).toEqual(false);
+      });
+    });
```

Comment:
I keep reading mixing `isCredentialsDataComplete` and `isCredentialsDetailsComplete` together :| Prob because I've been reading this file for several hours now, but at first glance I looked at this and was like "these two tests are exactly the same?..." 🤦‍♀️